clients/src/test/java/org/apache/kafka/clients/producer/MockProducerTest.java

```diff
@@ -77,6 +76,7 @@ public void testPartitioner() throws Exception {
         assertEquals("Partition should be correct", 1, metadata.get().partition());
         producer.clear();
         assertEquals("Clear should erase our history", 0, producer.history().size());
+        producer.close();
```

Comment:
Sure, that sounds good. I think this pattern could be applied in other places (files) too, but I'll skip them for now. I'll submit a new commit. Please let me know if you still see room for improvement. Thanks.